cluster-autoscaler/cloudprovider/mcm/mcm_manager.go

```diff
@@ -988,10 +988,10 @@ func buildGenericLabels(template *nodeTemplate, nodeName string) map[string]stri
 	return result
 }
 
-// isMachineTerminating returns true if machine is already being terminated or considered for termination by autoscaler.
-func isMachineTerminating(machine *v1alpha1.Machine) bool {
-	if !machine.GetDeletionTimestamp().IsZero() {
-		klog.Infof("Machine %q is already being terminated, and hence skipping the deletion", machine.Name)
+// isMachineFailedOrTerminating returns true if machine is already being terminated or considered for termination by autoscaler.
+func isMachineFailedOrTerminating(machine *v1alpha1.Machine) bool {
+	if !machine.GetDeletionTimestamp().IsZero() || machine.Status.LastOperation.State == v1alpha1.MachineStateFailed {
```

Comment:
```suggestion
	if !machine.GetDeletionTimestamp().IsZero() || machine.Status.CurrentStatus.Phase == v1alpha1.MachineFailed {
```

We need to check the phase and not the state of the last operation. The last operation, as the name suggests, will give info about the latest operation performed on the machine object. A failed machine is a machine whose Phase is Failed, not just the last Operation.